### ISSUE TITLE
New version of VSCode breaks the formatter (forge)

### ISSUE THREAD

**Issue by zgorizzo69** [juanfranblanco/vscode-solidity#399]
with forge 0.2.0 (https://github.com/foundry-rs/foundry/commit/93aa2d071b635a838030aa29ad85b52d361f5728 2023-03-11T00:09:06.544116259Z)

on linux with  VSCode 1.76.1 with JuanBlanco.solidity extension installed
in a mono repo with this .vscode/settings.json

```
{
    "editor.formatOnSave": true,
    "[solidity]": {
        "editor.defaultFormatter": "JuanBlanco.solidity" 
      },
    "solidity.compileUsingRemoteVersion": "v0.8.16",
    "solidity.formatter": "forge",
    "solidity.packageDefaultDependenciesContractsDirectory": "src",
    "solidity.packageDefaultDependenciesDirectory": "lib"
}
```
forge fmt works as expected
I have prettier installed at root 
but after vscode update the format on save doesn't work anymore
I tried opening only the packages/contracts folder in vsCode and also updating the vscode user's setting
It also doesn't work when I do a Format Document (forced)

**Comment by juanfranblanco** [comment]
@zgorizzo69 I have managed to replicate the issue, the solution was to reinstall forge as for some reason the path was not found by vscode.

**Comment by radeksvarz** [comment]
Indeed, the formatter is not listed
![image](https://github.com/juanfranblanco/vscode-solidity/assets/6020891/e6d57582-e547-4985-b8e0-0b460f941e99)

@juanfranblanco Is there any other way to fix?

**Comment by radeksvarz** [comment]
Really strange. It is listed now

![image](https://github.com/juanfranblanco/vscode-solidity/assets/6020891/f4f295d0-f640-45ee-8bb5-94aafd386b55)

But still does not format the document (upon save, but also upon direct Format command).

It looks like `forge fmt - ` is not called at all (or file not found).

I am using VSCodium in flatpack. Tried to enable `~/.foundry/bin/` in Flatseal.

Forge fmt works from the within the VSCodium terminal.

Is there any way to debug the invocation?

**Comment by outdoteth** [comment]
it works if i use prettier. when using forge the format on save is not working

**Comment by sergiimk** [comment]
This issue still exists. I created a new project with `forge init`, configured formatter the same way as described in the issue - everything was fine. After restarting VSCode formatter doesn't work any more. Had to switch from `"forge"` to `"prettier"`.

**Comment by radeksvarz** [comment]
For anyone having the same issue - this is the workaround:

CTRL-SHIFT-P Preferences: Open Keyboard Shortcuts (JSON)

add this:
```
  {
    "key": "ctrl+alt+f",
    "command": "workbench.action.terminal.sendSequence",
    "args": {
      "text": " forge fmt\u000D"
    }
  }
]
```

Mind the space before `forge` to avoid cluttering command history.

**Comment by parv3213** [comment]
I faced the same issues after the initial setup. Got resolved automatically after restarting VSCode.

**Comment by juanfranblanco** [comment]
Yes you will need to restart (sometimes) when changing the setting as is not picked up by vscode. As usual, is intermittent to make it more confusing to spot the problem, as other settings are ok.
